main.py

```diff
@@ -137,26 +140,79 @@ def _validate_validation_status(response: requests.Response) -> bool:
         sys.exit(1)
 
 
-def download_html_report(token: str, request_id: str, payload: Dict[str, Any]):
-    response = _retry_request(
+def download_report(token: str, request_id: str, payload: Dict[str, Any], response_type: str):
+    response_types = {"html": "text/html",
+                      "json": "application/json"}
+
+    return _retry_request(
         "GET",
         f"https://appinspect.splunk.com/v1/app/report/{request_id}",
-        headers={"Authorization": f"bearer {token}", "Content-Type": "text/html"},
+        headers={"Authorization": f"bearer {token}", "Content-Type": response_types[response_type]},
         data=payload,
     )
 
-    with open("AppInspect_response.html", "w") as f:
+
+def download_and_save_html_report(token: str, request_id: str, payload: Dict[str, Any]):
+    response = download_report(token, request_id, payload, "html")
+
+    with open(f"AppInspect_response.html", "w") as f:
         f.write(response.text)
 
 
+def read_bytes_response_as_dict(response):
+    temp = response.content
+
+    return json.loads(temp.decode("utf-8"))
```

Comment:
I am not sure you need to have a separate function for this.

Reply:
I created this function as I wanted to clean up the code a bit and move all this to a different file

Reply:
Let's leave this as it is for now